### PR TITLE
Make use of .Values.service.name

### PR DESCRIPTION
**Pull request by maratoid**
**What this PR does / why we need it**: Actually use the .Values.service.name value, with a default from template

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: vault-etcd/tls-generator/generate-certs.sh [hunk 1]
```diff
@@ -78,7 +78,7 @@ fi
 
 # statefulset name
 if [ -z ${GEN_STATEFULSET_NAME+x} ]; then
-    GEN_STATEFULSET_NAME="etcd-vault-etcd"
+    GEN_STATEFULSET_NAME="etcd-vault"
```
Comment (alejandroEsc) [inline vault-etcd/tls-generator/generate-certs.sh]: thanks!

File: vault-etcd/templates/statefulset.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@
 apiVersion: apps/v1beta1
 kind: StatefulSet
 metadata:
-  name: {{ template "vault-etcd.fullname" . }}
+  name: {{ .Values.service.name | default (include "vault-etcd.fullname" . ) }}
```
Comment (alejandroEsc) [inline vault-etcd/templates/statefulset.yaml]: why this change?

File: vault-etcd/values.yaml [hunk 3]
```diff
@@ -10,7 +10,7 @@ image:
   pullPolicy: "Always"
 
 service:
-  name: etcdvault
+  name: etcd-vault
```
Comment (alejandroEsc) [inline vault-etcd/values.yaml]: thanks!

File: vault-etcd/templates/statefulset.yaml (hunk 2 above)
Comment (maratoid) [reply vault-etcd/templates/statefulset.yaml]: to keep things consistent, and to match up with your script and cert names
